src/HdbDevice.cpp

```diff
@@ -345,6 +348,28 @@ void HdbDevice::get_hdb_signal_list(vector<string> & list)
 	if (!dev_prop[0].is_empty())
 		dev_prop[0]  >>  tmplist;
 
+	//use attribute list from file only if AttributeList property not present, or present with one empty line
+	if((tmplist.empty() || (tmplist.size() == 1 && tmplist[0].empty())) && !list_filename.empty())
+	{
+		string str;
+		std::ifstream in(list_filename);
+		if(in.is_open())
+		{
+			while (std::getline(in, str))
+			{
+			    if(!str.empty())
+				tmplist.push_back(str);
+			}
+			in.close();
+			if(!tmplist.empty())
+				list_from_file = true;
+		}
+		else
+		{
+			WARN_STREAM << __FUNCTION__<< ": cannot open Attribute List File '" << list_filename << "' error: '" << strerror(errno) << "'";
```

Comment:
I would even throw an exception here instead of logging a simple warning message.
Isn't this a real error if the AttributeListFile property was specified and this file cannot be read?

Reply:
Actually if you mean just throwing here an exception instead of the WARN_STREAM, it would end up being catched in the init_device. If you mean exiting the process, it could be OK, but at this point we should decide also how to behave if the file, opened here successfully for reading, cannot be opened for writing in put_signal_property.